src/org/labkey/test/pages/list/ConfirmDeletePage.java

```diff
@@ -0,0 +1,32 @@
+package org.labkey.test.pages.list;
+
+import org.labkey.test.Locator;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+
+public class ConfirmDeletePage extends LabKeyPage<ConfirmDeletePage.ElementCache>
+{
+    public ConfirmDeletePage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public BeginPage confirmDelete()
+    {
+        elementCache().deleteButton.click();
+        return new BeginPage(getDriver());
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends LabKeyPage.ElementCache
+    {
+        WebElement deleteButton = Locator.lkButton("Confirm Delete").findWhenNeeded(this);
```

Comment:
It might be worthwhile to either implement a waitForPage or put timeouts on elementCache elements; if there's a way to arrive at this page without waiting for a page refresh this will invite race conditions